hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java

```diff
@@ -1638,8 +1638,8 @@ private static class CompactionChecker extends ScheduledChore {
     @Override
     protected void chore() {
       for (Region r : this.instance.onlineRegions.values()) {
-        // Skip compaction if region is read only
-        if (r == null || r.isReadOnly()) {
+        // Skip compaction if region is read only or table disable compaction
```

Comment:
Let's be more clear:
`// If region is read only or compaction is disabled at table level, there's no need to iterate through region's stores`